docs/adr/0043-react-functional-tests.md

```diff
@@ -0,0 +1,170 @@
+- Signup: Full prod rollout was previously attempted and rolled back, second full prod rollout is planned for Train 283 (stage on 2024/04/24, production on 2024/05/01). New react tests were created, and where the signup flow was used in non-signup tests a condition was added with test steps to go through either flow based on feature flag. In preparation for the initial full prod rollout, backbone signup tests were skipped when the feature flag was on. By consequence, with the feature flag currently on for all environments, all backbone tests are skipped even though 85% of users see the backbone version of the signup flow.
+
+### Migration strategy and challenges
+
+Up until now, the primary strategy for supporting react flows with functional tests has mainly been to:
+
+- create new react-specific POMs (page object models)
+- duplicate tests and place them in a `react-conversion` directory (tests selected for duplication are based on the [Playwright Migration Analysis][2])
+- conditionally run these react tests based on a feature flag indicating the experiment is enabled for the route group
+
+Additional strategies that have been used include:
+
+- using conditional logic within tests to switch between react and backbone and react for a portion of the tests steps, based on feature flag configuration
+- using conditional logic within tests to run through both backbone and react versions of flows.
+- skip the backbone tests for full prod rollout (but there is no configuration value that can be checked)
+
+Additional challenges:
+
+- competing ideas have been implemented to bridge the gaps between React and Backbone in the functional tests
+- the desired end state of the react functional test migration is not universally agreed upon
+- Multiple react routes at different phases of rollout means that conditional logic is required to navigate between react/backbone test steps
+- The tests in `react-conversion` will need to be modified again during the cleanup phase to remove all of the feature flag check and experiment parameters
+
+### Questions
+
+- Are there new pattern that would simplify conversion moving forward (e.g., user-facing locators, single page POMs, route check in POMs, etc.)
+- Do we plan to discontinue all backbone tests? If so, when should we remove vs skip?
+- For tests that are not specific to the react conversion routes but that include test steps that use the converted routes, should we create new test files, or update the POM to be react/backbone agnostic?
+- How should we handle test setup (account creation and signup in fixtures) that currently use the backbone POMs?
+- How could we modify our test architecture to support coverage of both backbone and react tests while a route group experiment is enabled but not fully rolled out?
```

Comment:
It should be possible to force the backbone view via experiments, ie `?forceExperiment=generalizedReactApp&forceExperimentGroup=control`. If so then we might be able to just update the POM as needed.